src/main/java/amazin/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,50 @@
+package amazin.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public AuthenticationManager customAuthenticationManager() throws Exception {
+        return authenticationManager();
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests()
+                .antMatchers("/webjars/**", "/register", "/admin/register", "/forgot-password").permitAll()
+                .anyRequest().authenticated()
```

Comment:
The index also needs you to be authenticated? Why?

Reply:
are you saying we should allow people to be allowed to visit the homepage without logging in as anonymous users? If so this is a different issue and we can tackle this in a separate pr

Reply:
Yes they need to visit the homepage so they click the log in link. It is not a different issue. Not every user that uses our website need to type /login instead of clicking a login button on the homepage with limited access.

Reply:
The default behaviour of spring security is for the user to have to log in before doing anything else on the webpage. When the user is not logged in any url they try to access will redirect them automatically to the login page. For the feature of letting them also see the books and everything before logging in this could be done but I suggest this is done in a future PR